Prevent test failure and possible data loss due to API change in libical 3.0.7.

https://gitlab.gnome.org/GNOME/evolution-data-server/issues/185
https://lists.infradead.org/pipermail/libical-devel/2020-January/000907.html

Adapted from upstream:

https://gitlab.gnome.org/GNOME/evolution-data-server/commit/77384ab552c19bf374dbeda53dc37f98d07bd4ec

diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -108,7 +108,7 @@
 
 set(gcr_minimum_version 3.4)
 set(libgdata_minimum_version 0.15.1)
-set(libical_minimum_version 2.0)
+set(libical_minimum_version 3.0.7)
 set(libsecret_minimum_version 0.5)
 set(libxml_minimum_version 2.0.0)
 set(sqlite_minimum_version 3.7.17)
diff --git a/src/calendar/libedata-cal/e-cal-meta-backend.c b/src/calendar/libedata-cal/e-cal-meta-backend.c
index 7501f2a43dcc3af91e93e89751caf76e77c2ac23..93f4806ba2c5193f746e9fae5bf74d08ad05f8bf 100644
--- a/src/calendar/libedata-cal/e-cal-meta-backend.c
+++ b/src/calendar/libedata-cal/e-cal-meta-backend.c
@@ -3958,9 +3958,8 @@
 					gchar *base64;
 
 					base64 = g_base64_encode ((const guchar *) content, len);
-					new_attach = icalattach_new_from_data (base64, NULL, NULL);
+					new_attach = icalattach_new_from_data (base64, (GFunc) g_free, NULL);
 					g_free (content);
-					g_free (base64);
 
 					ecmb_remove_all_but_filename_parameter (prop);
 
